### PR TITLE
kernelci.org: add blog post about the Patchwork Integration

### PR DESCRIPTION
**Pull request by padovan**
Publish blog post about the under-development Patchwork Integration, so we can raise the awareness of the work and get people to follow it from now on.

### CODE REVIEW

File: kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Fortunately, we are not starting from scratch, as we already have a few examples of the Patchwork integration with CI in the BPF and netdev subsystems. (see the results for this [patch](https://patchwork.kernel.org/project/netdevbpf/patch/20231206095044.17844-1-duanqiangwen@net-swift.com/) for example).
+
+The big advantage of adding such support in KernelCI is that we can solve the problem for virtually all subsystems at the same time without reinventing and maintaining different CI infra across the board. 
+
+So recently, Nikolay Yurin, a Production Engineer from Meta, started developing the support [patchwork in the KernelCI](https://github.com/getpatchwork/patchwork/issues/557). It is still under heavy development and should reach a point that we can experiment with in early 2024.
+
+It is very important to put experimental support in place as we still don't know all the challenges we will face with the Patchwork integration, so engaging with the community to gather early feedback will be quite helpful. There are definitely corner cases and considerations that we will find out only when we start experimenting with it.
```
Comment (yurinnick) [inline kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md:22]: > started developing the support

I'd rather say suggested a way to implement :) So far I did nothing on patchwork side. However it would worth mentioning this kernelci/kernelci-api#307 and maybe kernelci/kernelci-pipeline#342

File: kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+date: 2023-12-12
+title: "Adding Patchwork integration to KernelCI"
+linkTitle: "Adding Patchwork integration to KernelCI"
+author: Gustavo Padovan
+description: >
+  Upcoming Patchwork integration in KernelCI aims helping maintainers reduce their workload, by deferring first line level of checks to KernelCI.
```
Comment (patersonc) [inline kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md:7]: Hi Gustavo,
Thank you for the MR.

Minor suggestion:
```suggestion
  Upcoming Patchwork integration in KernelCI aims to help maintainers reduce their workload, by deferring the first level of checks to KernelCI.
```

Comment (hardboprobot) [inline kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md:7]:
```suggestion
  Upcoming Patchwork integration in KernelCI aims to help maintainers reduce their workload by deferring first line level of checks to KernelCI.
```

File: kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Last month, at [Linux Plumbers](https://lpc.events/), most of the KernelCI Board Members were present and engaging with the community on several kernel testing related topics. One topic in particular has been grabbing the attention of some kernel maintainers: the ability to test patches out of Patchwork automatically. That way, maintainers can have a round of automated testing done before they put their hands on the patches.
+
+Testing the patches at the moment they arrive at the mailing list may be one of the best approaches because we can identify issues even before a person puts their eyes on the code. Developing a robust system to test new patches landing in Patchwork is not a trivial task. It will take some time to stabilize the support and deal with different corner cases. There is also the resource availability side, as doing that kind of testing at scale for several subsystems requires a lot of compute power. Last but not least, security is key, as blindly testing any patch that lands on the mailing list is not a great idea security-wise.
```
Comment (hardboprobot) [inline kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md:14]: Style suggestion: The previous line uses "put their hands on the patches", this one uses "puts their eyes on the code". Find an alternative phrasing for the latter.

File: kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Fortunately, we are not starting from scratch, as we already have a few examples of the Patchwork integration with CI in the BPF and netdev subsystems. (see the results for this [patch](https://patchwork.kernel.org/project/netdevbpf/patch/20231206095044.17844-1-duanqiangwen@net-swift.com/) for example).
+
+The big advantage of adding such support in KernelCI is that we can solve the problem for virtually all subsystems at the same time without reinventing and maintaining different CI infra across the board. 
+
+So recently, Nikolay Yurin, a Production Engineer from Meta, started developing the support [patchwork in the KernelCI](https://github.com/getpatchwork/patchwork/issues/557). It is still under heavy development and should reach a point that we can experiment with in early 2024.
```
Comment (hardboprobot) [inline kernelci.org/content/en/blog/posts/2023/patchwork-integration/index.md:20]:
```suggestion
So recently, Nikolay Yurin, a Production Engineer from Meta, started developing support for [patchwork in KernelCI](https://github.com/getpatchwork/patchwork/issues/557). It is still under heavy development and should reach a point that we can experiment with in early 2024.
```

**Comment by gctucker** [comment]
Mentioning Patchwork is kind of a "buzzword", it can attract some readers as it's well known but it would be fair to say that this feature is really about pre-merge testing in general and Patchwork is just one way of doing it on the client side. 
 The API side should be Patchwork-agnostic, there could be other bots discovering emails on mailing lists etc.  Also, this initial work is only about eBPF patches for now - which simplifies things quite a bit with some assumptions about things like base commits to apply the patches.

**Comment by padovan** [comment]
> Mentioning Patchwork is kind of a "buzzword", it can attract some readers as it's well known but it would be fair to say that this feature is really about pre-merge testing in general and Patchwork is just one way of doing it on the client side. The API side should be Patchwork-agnostic, there could be other bots discovering emails on mailing lists etc. Also, this initial work is only about eBPF patches for now - which simplifies things quite a bit with some assumptions about things like base commits to apply the patches.

Beyond a buzzword, it is the main way many kernel maintainers do their work. So telling the community that we want to serve that, whet some maintainers are already interested in working with us on that is important in my opinion.

Also, I agree this could eventually be a generic pre-merge testing feature, but right now there is only interest and effort available for the Patchwork part. And I believe it is in best interest of the project to make that integration happen.

**Comment by gctucker** [comment]
Yes, I mean it would seem useful to make it clearer that the API is not going to be tied to Patchwork.  It's only a starting point with the BPF use-case.

**Comment by padovan** [comment]
Let me add a note about that in the post then. I think it is important to highlight that we want to solve pre-merge in general too.

**Comment by gctucker** [comment]
@padovan Hello, is this still being worked on or has the blog post idea been abandoned for now?

**Comment by padovan** [comment]
> @padovan Hello, is this still being worked on or has the blog post idea been abandoned for now?

We are still working on it, but I believe we have to wait a bit more to publish it.
